packages/btcindexer/src/btcindexer.ts

```diff
@@ -30,12 +30,35 @@ interface BlockRecord {
 	block_height: number;
 }
 
-interface Storage {
+export interface Storage {
 	d1: D1Database; // SQL DB
 	blocksDB: KVNamespace;
 	nbtcTxDB: KVNamespace;
 }
 
+export type NbtcTxStatus = "confirming" | "finalized" | "minted" | "failed" | "reorg";
+
+export interface TransactionStatusResponse {
+	btc_tx_id: string;
+	status: NbtcTxStatus;
+	block_height: number | null;
+	confirmations: number;
+	sui_recipient: string;
+	amount_sats: number;
+}
+
+interface NbtcTxD1Row {
+	tx_id: string;
+	block_hash: string;
+	block_height: number | null;
+	vout: number;
+	sui_recipient: string;
+	amount_sats: number;
+	status: string;
+	created_at: string;
+	updated_at: string;
```

Comment:
the two above should be timestamp (number)

Reply:
schema ```    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP``` generates string

Reply:
TIMESTAMP is not a proper datatype in sqlite3 - it's [from sqlite2](https://www.sqlite.org/datatypes.html).
In sqlite3 we have the following option: https://www.sqlite.org/datatype3.html#date_and_time_datatype

For default try to change `DEFAULT unixepoch('subsec'),`